src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,200 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from './VisualizerDefault'
+import {natlog} from '../shared/math'
+
+/** md
+# Factor Histogram
+
+(example image should go here)
+
+This visualizer takes the prime factorization of each element in the sequence
+and then creates a histogram of the number of elements of the sequence
+with X prime factors. the number of elements with X prime factors is along
+the x-axis. And the height is how many elements are in each bin of X.
+
+## Parameters
+**/
+
+class FactorHistogramVisualizer extends VisualizerDefault {
+    name = 'Histogram'
+
+    binSize = 1
+    terms = 100
+    firstIndex = this.seq.first
+
+    params = {
+        binSize: {
+            value: this.binSize,
+            forceType: 'integer',
+            displayName: 'Bin Size',
+            required: true,
+        },
+
+        firstIndex: {
+            value: this.firstIndex,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: true,
+        },
+
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'How many terms of the series',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Obtain the true first index
+    startIndex(): number {
+        return Math.max(this.firstIndex, this.seq.first)
+    }
+
+    // Obtain the true number of terms
+    numTerms(): number {
+        return Math.min(this.terms + this.startIndex(), this.seq.last)
+    }
```

Comment:
Isn't this in the end the last index, not the number of terms? Suggest renaming.